chain/network/src/peer/rate_counter.rs

```diff
@@ -1,86 +1,93 @@
-// Copyright 2018 The Grin Developers
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 use std::collections::VecDeque;
-use std::time::{Duration, SystemTime};
-
-const MINUTE: Duration = Duration::from_secs(60);
+use std::time::{Duration, Instant};
 
-/// Stores list of entries for `RateCounter`
-struct Entry {
-    /// bytes since last reset
+/// Represents a single event in time.
+struct Event {
+    /// Time
+    instant: Instant,
+    /// Number of bytes
     bytes: u64,
-    /// Time we created the entry.
-    recorded: SystemTime,
 }
 
-/// A rate counter tracks number of transfers, the amount of data exchanged and the rate of transfer
-/// over the last minute.
-pub struct RateCounter {
-    entries: VecDeque<Entry>,
-    bytes_sum: u64,
+/// Represents all events which happened in last minute.
+#[derive(Default)]
+pub struct TransferStats {
+    /// We keep list of entries not older than 1m.
+    events: VecDeque<Event>,
+    /// Sum of bytes for all entries.
+    bytes_per_min: u64,
```

Comment:
nit; I'd not call it 'per_min' - maybe bytes_in_events? or something similar?

Reply:
I renamed to `total_bytes_in_events`.